azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -86,28 +87,53 @@ public int createImageType(ImageType imageType) {
       /* what will happen if there is a partial failure in
          any of the below statements?
          Ideally all should happen in a transaction */
-      imageTypeId = databaseOperator.transaction(insertAndGetSpaceId);
+      imageTypeId = this.databaseOperator.transaction(insertAndGetSpaceId);
       log.info("Created image type id :" + imageTypeId);
-    } catch (SQLException e) {
+    } catch (final SQLException e) {
       log.error("Unable to create the image type metadata", e);
       throw new ImageMgmtDaoException("Exception occurred while creating image type metadata");
     }
     return imageTypeId;
   }
 
   @Override
-  public Optional<ImageType> getImageTypeByName(String name) {
-    FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+  public Optional<ImageType> getImageTypeByName(final String name) {
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
     List<ImageType> imageTypes = new ArrayList<>();
     try {
-      imageTypes = databaseOperator
+      imageTypes = this.databaseOperator
           .query(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_NAME, fetchImageTypeHandler, name);
       // Check if there are more then one image types for a given name. If so throw exception
       if (imageTypes != null && imageTypes.size() > 1) {
         throw new ImageMgmtDaoException("Can't have more that one image type record for a given "
             + "type with name : " + name);
       }
-    } catch (SQLException ex) {
+    } catch (final SQLException ex) {
+      log.error(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_NAME + " failed.", ex);
+      throw new ImageMgmtDaoException(
+          "Unable to fetch image type metadata from image type : " + name);
+    }
+    return imageTypes.isEmpty() ? Optional.empty() : Optional.of(imageTypes.get(0));
+  }
+
+  @Override
+  public Optional<ImageType> getImageTypeWithOwnershipsByName(final String name)
+      throws ImageMgmtException {
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+    List<ImageType> imageTypes = new ArrayList<>();
+    try {
+      imageTypes = this.databaseOperator
+          .query(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_NAME, fetchImageTypeHandler, name);
+      // Check if there are more then one image types for a given name. If so throw exception
+      if (imageTypes != null && imageTypes.size() > 1) {
+        throw new ImageMgmtDaoException("Can't have more that one image type record for a given "
+            + "type with name : " + name);
+      }
+      if (!imageTypes.isEmpty()) {
+        final ImageType imageType = imageTypes.get(0);
+        imageType.setOwnerships(getImageOwnerships(imageType.getName()));
```

Comment:
Here you are executing two queries. One to get image type and second to get Ownership. You should write single query to fetch both image type and ownership. You are anyway joining with Image type table in getImageOwnerships. That would suffice. You can manage the records in Handler itself and create ImageType object properly with ownership information.

Reply:
One image type has more than one owners. It is more of a parent child relationships. Hence, from image_types table only one row will be fetched. From image_ownerships table more that one row will be fetched for the image types. And the ownerships records will be added as a list to the image type. Hence, both the queries can't be combined.